components/places/src/error.rs

```diff
@@ -71,6 +71,11 @@ pub enum ErrorKind {
 
     #[fail(display = "Error parsing URL: {}", _0)]
     UrlParseError(#[fail(cause)] url::ParseError),
+
+    // Maybe we should try to fabricate a rusqlite::Error that looks like the
+    // interrupted error?
```

Comment:
Or, can we go the other way and have our `From<rusqlite::Error>` impl convert `ErrorCode::OperationInterrupted` into `InterruptedError`, and others into `SqlError`? Looking at your autocomplete example changes, and your note in `get_code`, I agree; it's unfortunate that we need to handle interruption in two places...

Reply:
That's probably a cleaner approach, but it's a bit of a pain to do, and doing so will also cause us to lose backtrace info about which op was interrupted. I'll think about this, but for now I'd prefer to keep as-is.